torchgen/gen.py

```diff
@@ -2297,8 +2297,17 @@ def gen_op_headers(
             if g1.view_copy is None or g2.view_copy is None:
                 continue
             # TODO: make this more first class in the data model
-            same_base_op = str(g1.view_copy.func.name.name) == str(
-                g2.view_copy.func.name.name
+            g1_base_name = str(g1.view_copy.func.name.name)
+            g2_base_name = str(g2.view_copy.func.name.name)
+            g1_stripped_schema = str(g1.view_copy.func.arguments).replace(
+                "SymInt", "int"
+            )
+            g2_stripped_schema = str(g2.view_copy.func.arguments).replace(
+                "SymInt", "int"
+            )
```

Comment:
I think all of these methods can probably live directly in `model.py`, since they're acting directly on `Argument` and `Type` which also live there.

Hmm one way to do it would be to define a `sym_int_to_int` method on each of the `Type` classes in `model.py`. Something like:
```
class BaseType(Type):
    ...
    def sym_int_to_int(self) -> "BaseType":
        if self.name == BaseTy.SymInt:
            return BaseType(BaseTy.int)
        return self
...
class ListType(Type):
    ...
    def sym_int_to_int(self) -> "ListType":
        return ListType(self.elem.sym_int_to_int(), self.size)

... (same thing for `OptionalType`)
```